### PR TITLE
feat(NODE-3419): define MongoRuntimeError children

### PR DESCRIPTION
**Pull request by andymina**
## Description
- Added class definitions of MongoRuntimeError children as seen in the second and third graph of [docs/errors.md](https://github.com/mongodb/node-mongodb-native/blob/4.0/docs/errors.md) and needed in [NODE-3404](https://jira.mongodb.org/browse/NODE-3404) and [NODE-3405](https://jira.mongodb.org/browse/NODE-3405).

**What changed?**
Classes added:
- MongoURIError
- MongoCompressionError
- MongoDecompressionError
- MongoTransactionError
- MongoExpiredSessionError
- MongoKerberosError
- MongoStreamError
- MongoChangeStreamError
- MongoGridFSStreamError
- MongoGridFSChunkError
- MongoResourceClosedError
- MongoServerClosedError
- MongoStreamClosedError
- MongoTopologyClosedError

### CODE REVIEW

**Review by W-A-James — APPROVED** [review]
LGTM 👍🏿

File: src/error.ts [hunk 1]
```diff
@@ -198,6 +198,240 @@ export class MongoRuntimeError extends MongoDriverError {
...
+    super(message);
+  }
+
+  get name(): string {
+    return 'MongoCompressionError';
+  }
+}
+
+/**
+ * An error generated when teh driver fails to decompress
```
Comment (emadum) [inline src/error.ts:235]:
```suggestion
 * An error generated when the driver fails to decompress
```

File: src/error.ts [hunk 2]
```diff
@@ -198,6 +198,240 @@ export class MongoRuntimeError extends MongoDriverError {
...
+    super(message);
+  }
+
+  get name(): string {
+    return 'MongoGridFSChunkError';
+  }
+}
+
+/**
+ * An error generated when an attempt to access a resource
```
Comment (emadum) [inline src/error.ts:368]:
```suggestion
 * An error generated when an attempt is made to access a resource
```

**Review by emadum — CHANGES_REQUESTED** [review]
I think this documentation would read better if we replaced the boilerplate `An error generated when an attempt is made to ...` to something more succinct like `Thrown when attempting to ...`. Could we also make the line length used for wrapping consistent across all the docs?

**Review by nbbeeken — APPROVED** [review]
Seconded on Eric's comments, otherwise LGTM

**Review by emadum — APPROVED** [review]
LGTM 👍

**Review by dariakp — APPROVED** [review]
LGTM

**Pull request merged**